server/src/tests/integration.rs

```diff
@@ -81,6 +91,15 @@ async fn test_on_ethereum() -> Result<()> {
     deploy_move_counter().await?;
 
     // 12. Cleanup generated files and folders
+    hb.abort();
```

Comment:
Maybe we can move the heartbeat cleanup logic into `heartbeat.rs` and call it with a single line here.